lib/ecto/adapters/mssql/connection.ex

```diff
@@ -0,0 +1,1568 @@
+      {expr || paren_expr(source, sources, query), name}
+    end
+
+    defp quote_name(name) when is_atom(name) do
+      quote_name(Atom.to_string(name))
+    end
+
+    defp quote_name(name) do
+      if String.contains?(name, ["[", "]"]) do
+        error!(nil, "bad field name #{inspect(name)} '[' and ']' are not permited")
+      end
+
+      "[#{name}]"
+    end
+
+    defp quote_table(nil, name), do: quote_table(name)
+
+    defp quote_table(prefix, name),
+      do: Enum.map_join([quote_table(prefix), ".", quote_table(name)], "", &"#{&1}")
+
+    defp quote_table(name) when is_atom(name), do: quote_table(Atom.to_string(name))
+
+    defp quote_table(name) do
+      if String.contains?(name, "[") or String.contains?(name, "]") do
+        error!(nil, "bad table name #{inspect(name)} '[' and ']' are not permited")
+      end
+
+      "[#{name}]"
+    end
+
+    defp unquoted_name(prefix, name, column_name),
+      do: unquoted_name(unquoted_name(prefix, name), column_name)
+
+    defp unquoted_name(nil, name), do: unquoted_name(name)
+
+    defp unquoted_name(prefix, name) do
+      prefix = if is_atom(prefix), do: Atom.to_string(prefix), else: prefix
+      name = if is_atom(name), do: Atom.to_string(name), else: name
+
+      [prefix, ".", name]
+    end
+
+    defp unquoted_name(name) when is_atom(name), do: unquoted_name(Atom.to_string(name))
+
+    defp unquoted_name(name) do
+      if String.contains?(name, ["[", "]"]) do
+        error!(nil, "bad table name #{inspect(name)} '[' and ']' are not permited")
+      end
+
+      name
+    end
+
+    defp intersperse_map([], _separator, _mapper), do: []
+    defp intersperse_map([elem], _separator, mapper), do: mapper.(elem)
+
+    defp intersperse_map([elem | rest], separator, mapper) do
+      [mapper.(elem), separator | intersperse_map(rest, separator, mapper)]
+    end
+
+    defp intersperse_reduce(list, separator, user_acc, reducer, acc \\ [])
+
+    defp intersperse_reduce([], _separator, user_acc, _reducer, acc),
+      do: {acc, user_acc}
+
+    defp intersperse_reduce([elem], _separator, user_acc, reducer, acc) do
+      {elem, user_acc} = reducer.(elem, user_acc)
+      {[acc | elem], user_acc}
+    end
+
+    defp intersperse_reduce([elem | rest], separator, user_acc, reducer, acc) do
+      {elem, user_acc} = reducer.(elem, user_acc)
+      intersperse_reduce(rest, separator, user_acc, reducer, [acc, elem, separator])
+    end
+
+    defp if_do(condition, value) do
+      if condition, do: value, else: []
+    end
+
+    defp escape_string(value) when is_binary(value) do
+      value
+      |> :binary.replace("'", "''", [:global])
+    end
+
+    defp ecto_to_db(type, size, precision, scale, query \\ nil)
+
+    defp ecto_to_db({:array, _}, _, _, _, query),
+      do: error!(query, "Array type is not supported by TDS")
+
+    defp ecto_to_db(:id, _, _, _, _), do: "bigint"
+    # defp ecto_to_db(:integer, _, _, _, _), do: "bigint"
```

Comment:
```suggestion
    defp ecto_to_db(:integer, _, _, _, _), do: "int"
```

Reply:
this will cause bug in schema_migrations query `type(version, :integer)` , error is arithmetic overflow, sql will try to convert bigint into integer

Reply:
Even after the fix on this [line](https://github.com/elixir-ecto/ecto_sql/pull/184/files#diff-e9f0383caccf95b7b730c86af44ed857R823)?